pdfbox/src/main/java/org/apache/pdfbox/pdfparser/COSParser.java

```diff
@@ -2066,4 +2075,36 @@ protected COSBase parseTrailerValuesDynamically(COSDictionary trailer) throws IO
     public COSDictionary getLastTrailer() {
         return xrefTrailerResolver.getLastTrailer();
     }
+
+    protected void isLinearized(Long fileLen) throws IOException {
+        Long offset = fileLen;
+        Map.Entry<COSObjectKey, Long> object = null;
+        for (Map.Entry<COSObjectKey, Long> entry : document.getXrefTable().entrySet()) {
+            if (offset.compareTo(entry.getValue()) > 0) {
+                object = entry;
+                offset = entry.getValue();
+            }
+        }
+
+        final COSObject pdfObject = new COSObject(null);
+        if (object != null) {
+            parseFileObject(object.getValue(),
+                    object.getKey(),
+                    object.getKey().getNumber(),
+                    object.getKey().getGeneration(),
+                    pdfObject);
+        } else throw new IllegalStateException("Document has no any object.");
```

Comment:
"Document has no any object." -> "Linearization dictionary is missed in document"